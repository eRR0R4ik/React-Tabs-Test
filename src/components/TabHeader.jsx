import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Tab from './Tab.jsx';


class TabHeader extends Component {
  render() {
    let activeClass = this.props.activeId;

    /**
     * render Tab for each element of data
     */
    let tabs = this.props.data.map((item, index) => {
      return (
        <Tab
          key={`content ${item.id}`}
          activeClass={`phase-${index + 1} ${activeClass === index ? '__active' : ''}`}
          name={item.name}
          index={index}
          onClick={this.props.onClick}
        />
      )
    });

    return (
      <ul className="phases-tabs">{tabs}</ul>
    )
  }
}

TabHeader.propTypes = {
  onClick: PropTypes.func.isRequired,
  activeId: PropTypes.number.isRequired,
  data: PropTypes.array.isRequired,
};

export default TabHeader;
