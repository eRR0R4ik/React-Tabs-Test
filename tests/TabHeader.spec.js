import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TabHeader from '../src/components/TabHeader.jsx';

describe('Component: TabHeader', () => {
  const testFunction = () => true;
  const minProps = {
    data: [
      {
        picture: 'mockPics1',
        name: 'mockName1',
        index: 0
      },
      {
        picture: 'mockPics2',
        name: 'mockName2',
        index: 1
      },
    ],
    onClick: testFunction,
    activeId: 1
  }
  it('correctly pass props into TabHeader', () => {
    expect(shallow(<TabHeader {...minProps} />).length).toEqual(1);
  });

  it('should render correctly', () => {
    const component = shallow(
      <TabHeader {...minProps} />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders an `.__active`', () => {
    const component = shallow(<TabHeader {...minProps} />);
    expect(component.find('.__active').length).toEqual(0);
  });
});
