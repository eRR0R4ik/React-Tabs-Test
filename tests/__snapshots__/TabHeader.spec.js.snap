// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TabHeader should render correctly 1`] = `
<ul
  className="phases-tabs"
>
  <Tab
    activeClass="phase-1 "
    index={0}
    key="content undefined"
    name="mockName1"
    onClick={[Function]}
  />
  <Tab
    activeClass="phase-2 __active"
    index={1}
    key="content undefined"
    name="mockName2"
    onClick={[Function]}
  />
</ul>
`;
